### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by yosriz**
add JVM unit tests,  about 70% coverage now 
* related `Request` area cannot be tested in JVM
* `KeyStore` is a fake for now, so no tests written yet
* this change depends on https://github.com/kinfoundation/android-stellar-sdk/pull/6

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -1,26 +1,34 @@
...
 
     supportDependencies = [
-            appCompat: buildDependency(supportPackage, 'appcompat-v7', supportVersion),
+            supportAnnotations : buildDependency(supportPackage, 'support-annotations', supportVersion)
     ]
 
     testingDependencies = [
             junit         : buildDependency(junitPackage, 'junit', junitVersion),
-            mockitoAndroid: buildDependency(mockitoPackage, "mockito-android", mockitoVersion),
+            mockitoAndroid: buildDependency(mockitoPackage, "mockito-android", mockitoAndroidVersion),
```
Comment (OrenZak) [inline dependencies.gradle]: can you replace `"` with `'` to all?

File: dependencies.gradle [hunk 2]
```diff
@@ -1,26 +1,34 @@
 ext {
 
     //Version
-    supportVersion = '26.+'
+    supportVersion = '27.0.2'
```
Comment (OrenZak) [inline dependencies.gradle]: I'm using `26.1.0` can we both use the same version?

File: kin-sdk-core/build.gradle [hunk 3]
```diff
@@ -31,12 +31,15 @@ android {
 }
 
 dependencies {
-    implementation 'com.github.kinfoundation:android-stellar-sdk:master-SNAPSHOT'
-    implementation supportDependencies.appCompat
+    implementation 'com.github.kinfoundation:android-stellar-sdk:transaction_increase_timeout_fix_error_handling-SNAPSHOT'
```
Comment (OrenZak) [inline kin-sdk-core/build.gradle]: why did you change it to: `transaction_increase_timeout_fix_error_handling`
I would also add it to `dependencies.gradle`

File: kin-sdk-core/src/main/java/kin/sdk/core/AccountActivator.java [hunk 4]
```diff
@@ -56,6 +51,21 @@ void activate(@NonNull Account account, @NonNull String passphrase) throws Opera
         }
     }
 
+    private void verifyParams(@NonNull Account account, @NonNull String passphrase) {
```
Comment (OrenZak) [inline kin-sdk-core/src/main/java/kin/sdk/core/AccountActivator.java:54]: You can use as `verifyParams(@NonNull Object... objs)` and do a for loop of checknotnull

File: kin-sdk-core/build.gradle (hunk 3 above)
Comment (yosriz) [reply kin-sdk-core/build.gradle]: I have PR for the stellar-sdk, let's review it together with this one, so it will get to master/dev together

File: kin-sdk-core/src/main/java/kin/sdk/core/AccountActivator.java (hunk 4 above)
Comment (yosriz) [reply kin-sdk-core/src/main/java/kin/sdk/core/AccountActivator.java:54]: I won't be able to infer the param name if I switch to loop (I need it for proper exception message)

File: kin-sdk-core/src/main/java/kin/sdk/core/BalanceQuery.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import kin.sdk.core.exception.AccountNotActivatedException;
+import kin.sdk.core.exception.OperationFailedException;
+import org.stellar.sdk.KeyPair;
+import org.stellar.sdk.Server;
+import org.stellar.sdk.responses.AccountResponse;
+import org.stellar.sdk.responses.HttpResponseException;
+
+class BalanceQuery {
+
+
```
Comment (berryve) [inline kin-sdk-core/src/main/java/kin/sdk/core/BalanceQuery.java]: extra line

File: kin-sdk-core/src/main/java/kin/sdk/core/TransactionSender.java [hunk 6]
```diff
@@ -129,12 +143,18 @@ private AccountResponse loadSourceAccount(@NonNull KeyPair from) throws Operatio
...
+            if (response == null) {
+                throw new OperationFailedException("can't get transaction response");
+            }
             if (response.isSuccess()) {
                 return new TransactionIdImpl(response.getHash());
             } else {
                 throw Utils.createTransactionException(response);
             }
-        } catch (IOException e) {
+        } catch (
```
Comment (berryve) [inline kin-sdk-core/src/main/java/kin/sdk/core/TransactionSender.java]: format

File: kin-sdk-core/src/main/java/kin/sdk/core/Utils.java [hunk 7]
```diff
@@ -10,7 +10,7 @@
 final class Utils {
 
     private Utils() {
-
+        //not instances
```
Comment (berryve) [inline kin-sdk-core/src/main/java/kin/sdk/core/Utils.java]: no

File: kin-sdk-core/src/test/java/kin/sdk/core/AccountActivatorTest.java [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+
+        assertThat(mockWebServer.takeRequest().getRequestUrl().toString(), containsString(ACCOUNT_ID_FROM));
+        assertThat(mockWebServer.takeRequest().getBody().readUtf8(), equalTo(TX_BODY));
+    }
+
+    @Test
+    public void activate_HasTrustAlready_NoChangeTrust() throws Exception {
+        mockWebServer.enqueue(generateSuccessMockResponse(this.getClass(), "activate_account.json"));
+
+        accountActivator.activate(account, "");
```
Comment (berryve) [inline kin-sdk-core/src/test/java/kin/sdk/core/AccountActivatorTest.java:118]: check that there is no change trust call

File: kin-sdk-core/src/test/java/kin/sdk/core/ClientWrapperTest.java [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+        fakeServiceProvider = new ServiceProvider("", ServiceProvider.NETWORK_ID_TEST);
+        fakeKeyStore = new FakeKeyStore();
+
+        clientWrapper = new ClientWrapper(fakeServiceProvider, fakeKeyStore, mockTransactionSender,
+            mockAccountActivator, mockBalanceQuery);
+    }
+
+    @Test
+    public void wipeoutAccount() throws Exception {
+        //TODO
```
Comment (berryve) [inline kin-sdk-core/src/test/java/kin/sdk/core/ClientWrapperTest.java:41]: make sure test is added

**Review by berryve — APPROVED** [review]

File: kin-sdk-core/src/test/java/kin/sdk/core/ClientWrapperTest.java (hunk 9 above)
Comment (yosriz) [reply kin-sdk-core/src/test/java/kin/sdk/core/ClientWrapperTest.java:41]: will added at a different task
